sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileServiceAsyncApiTests.java

```diff
@@ -260,19 +272,21 @@ public void setAndGetProperties() {
         ShareServiceProperties updatedProperties
             = new ShareServiceProperties().setHourMetrics(metrics).setMinuteMetrics(metrics).setCors(new ArrayList<>());
 
-        StepVerifier.create(primaryFileServiceAsyncClient.getPropertiesWithResponse()).assertNext(it -> {
-            FileShareTestHelper.assertResponseStatusCode(it, 200);
-            assertTrue(FileShareTestHelper.assertFileServicePropertiesAreEqual(originalProperties, it.getValue()));
-        }).verifyComplete();
-
-        StepVerifier.create(primaryFileServiceAsyncClient.setPropertiesWithResponse(updatedProperties))
-            .assertNext(it -> FileShareTestHelper.assertResponseStatusCode(it, 202))
-            .verifyComplete();
-
-        StepVerifier.create(primaryFileServiceAsyncClient.getPropertiesWithResponse()).assertNext(it -> {
-            FileShareTestHelper.assertResponseStatusCode(it, 200);
-            assertTrue(FileShareTestHelper.assertFileServicePropertiesAreEqual(updatedProperties, it.getValue()));
-        }).verifyComplete();
+        StepVerifier.create(originalPropertiesMono.flatMap(
```

Comment:
To use the zipped originalProperties , you can make your first StepVerifier look something like this: 

```
StepVerifier.create(originalProperties)
            .assertNext(tuple -> {
                FileShareTestHelper.assertResponseStatusCode(tuple.getT2(), 200);
                assertTrue(FileShareTestHelper.assertFileServicePropertiesAreEqual(tuple.getT1(), tuple.getT2().getValue()));
            })
            .verifyComplete();
```